opta/commands/secret.py

```diff
@@ -39,9 +35,42 @@
     show_default=True,
 )
 
+module_option = click.option(
+    "-m", "--module", default=None, help="The module to get the secret for"
+)
+
+
+def get_secret_name_and_namespace(
+    layer: Layer, module_name: Optional[str]
+) -> Tuple[str, str]:
+    k8s_services = layer.get_module_by_type("k8s-service")
+    helm_charts = layer.get_module_by_type("helm-chart")
+    total_modules = k8s_services + helm_charts
+    if module_name is None and len(total_modules) > 1:
+        module_name = click.prompt(
+            "Multiple k8s-service/helm chart modules found. Please specify which one do you want the secret for.",
+            type=click.Choice([x.name for x in total_modules]),
+        )
+    if module_name is None:
+        module: Module = total_modules[0]
+    else:
+        module = layer.get_module(module_name)  # type: ignore
+        if module is None:
+            raise UserErrors(
+                f"Could not find helm/k8s-service module with name {module_name}"
+            )
```

Comment:
```suggestion
        try:
            module = next(module for module in total_modules if module.name == module_name)
        except StopIteration:
            raise UserErrors(
                f"Could not find helm/k8s-service module with name {module_name}"
            ) from None
```

Disabling type checking should be avoided wherever possible.